cmd/telemetrygen/internal/common/config.go

```diff
@@ -81,11 +81,12 @@ func (c *Config) CommonFlags(fs *pflag.FlagSet) {
 
 	// custom headers
 	c.Headers = make(map[string]string)
-	fs.Var(&c.Headers, "otlp-header", "Custom header to be passed along with each OTLP request. The value is expected in the format key=value."+
+	fs.Var(&c.Headers, "otlp-header", "Custom header to be passed along with each OTLP request. The value is expected in the format key=value. "+
 		"Flag may be repeated to set multiple headers (e.g -otlp-header key1=value1 -otlp-header key2=value2)")
 
 	// custom resource attributes
 	c.ResourceAttributes = make(map[string]string)
-	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=\"value\"."+
+	fs.Var(&c.ResourceAttributes, "otlp-attributes", "Custom resource attributes to use. The value is expected in the format key=\"value\". "+
+		"Value can contain special syntax: ${random:lower-upper} that will be replaces with value in range [lower, upper)."+
```

Comment:
The `${...}` syntax is a bit cumbersome to use in most shells. Both in PowerShell and Bash (I assume probably also in zsh and Fish), you need to escape this with single quotes as in `telemetrygen traces -otlp-attributes '"${random:0-100}"'`, doing something like `telemetrygen traces -otlp-attributes ${random:0-100}` or `telemetrygen traces -otlp-attributes "${random:0-100}"` won't work.

Is there any other syntax that would work here?

Reply:
I agree, that using `$` is not the best. How about using `[]`, so something like: `[random-int:0-100]`?